### PR TITLE
Mega Menu: Remove h2 heading wrapper around trigger button

### PR DESCRIPTION
**Pull request by anselmbradford**
Seems kind of weird that we wrap the hamburger button in an `h2` and reset the default styles for `h2`. I think this was probably originally added in anticipation of more content being in this part of the menu that didn't show up.

**I also wondered if there's an accessibility reason to have heading content in a `nav`, but it seems to work the same with or without with voiceover. Anyone else know of any reason the heading wrapper may be needed?**

## Removals

- Remove h2 `o-mega-menu_heading` heading wrapper around mobile trigger button.

## Changes

- Adds h2 heading wrapper around top-level menu links.

## Testing

1. Pull branch and build.
2. Visit the mobile version of the site and expand/collapse the hamburger menu. It should work just like production.

### CODE REVIEW

**Comment by anselmbradford** [comment]
@contolini

**Comment by Scotchester** [comment]
The heading was added in #4228. I'd take a look at the description of that PR and maybe discuss with @cfarm, but it seems to me like the logic for including it was sound at the time. Possibly some of the structural changes in the new MM change the situation though?

**Comment by contolini** [comment]
Per [W3C](https://www.w3.org/TR/wai-aria-practices/#menubutton) I'm okay removing it. We need to add `role="menu"` to the [wrapper](https://github.com/cfpb/cfgov-refresh/blob/master/cfgov/jinja2/v1/_includes/organisms/mega-menu.html#L111) though.

**Comment by anselmbradford** [comment]
Are you aware of any resources saying we should include the menu in the headings list in the rotor? I visited a few sites with a hamburger menu (washingtonpost, nytimes, whitehouse, usps) and none of them include the menu in the headings list. It's clearest on the news sites where the headings list is just a list of news article headings.

**Comment by contolini** [comment]
Per [W3C](https://www.w3.org/TR/wai-aria-practices/#menubutton) I'm okay removing it. We need to add `role="menu"` to the [wrapper](https://github.com/cfpb/cfgov-refresh/blob/master/cfgov/jinja2/v1/_includes/organisms/mega-menu.html#L111) though.

**Comment by Scotchester** [comment]
> Are you aware of any resources saying we should include the menu in the headings list in the rotor? I visited a few sites with a hamburger menu (washingtonpost, nytimes, whitehouse, usps) and none of them include the menu in the headings list. It's clearest on the news sites where the headings list is just a list of news article headings.

From a quick survey of some search results for "accessible mega menu", it seems like some folks recommend using them (without saying they are _necessary_), while others don't. Seems like a judgement call, and one that we've already made. I'd like Cat to weigh in before reversing that.

[Sidenote: I checked out the homepage's heading structure in the WAVE toolbar and some of the recent changes have introduced some issues that we should probably address.]

**Comment by cfarm** [comment]
I didn't see @anselmbradford address this, which is from [my PR](https://github.com/cfpb/cfgov-refresh/pull/4228) that introduced the h2

> gives a parent heading element to the new <h3> headings that I added for overview text, and to the existing <h4> elements (see details below)

that's the main reason for the change and other decisions were to make sure that the heading hierarchy is sensible which makes it accessible for screen reader users. having heading elements gives screen readers natural landmarks to read aloud to users.

if there are still h3s in this menu then I would not approve this change. that's why the h2 was added, because those were showing up in the headings list on screen readers but the hierarchy made no sense without knowing the context and making the menu navigable from the heading list too.

**Comment by anselmbradford** [comment]
> gives a parent heading element to the new h3 headings that I added for overview text, and to the existing h4 elements

This makes sense, but the h2 is not a parent of the h3s and h4s. It's enclosing the hamburger menu item (aka the "menu trigger"), which is a sibling of the main menu content. It seems like the h2 should be surrounding the top-level overview link, not the menu trigger. So "Consumer Tools" in this image:

Screen Shot 2020-04-20 at 12 16 14 PM

File: cfgov/unprocessed/css/organisms/mega-menu.less [hunk 1]
```diff
@@ -365,6 +365,13 @@ body {
         margin-top: 0;
     }
 
+    // Reset default h2 heading styles.
+    h2 {
+        font-size: 1em;
+        line-height: 22px;
```
Comment (Scotchester) [inline cfgov/unprocessed/css/organisms/mega-menu.less:371]: Can we make this unitless?

```suggestion
        line-height: 1.375;
```

**Pull request merged**